### ISSUE TITLE
New TSQL rule suggestion: Flag statements that do not end with semi-colon

### ISSUE THREAD

**Issue by jpers36** [sqlfluff/sqlfluff#1748]
The semi-colon statement terminator is optional but best practice.

This could be fixed automatically.

Anti-pattern:

```
SELECT 1
```

Pattern:

```
SELECT 1;
```

This could potentially apply to other dialects as well; I am not familiar enough with other dialects to say for sure.

An exception to the above rule would be the BEGIN statement, where we should flag/remove the optional semi-colon if we see it.
